src/App.jsx

```diff
@@ -1,17 +1,24 @@
-import "./App.css";
+import { useState } from "react";
 import { Route, Routes } from "react-router-dom";
+import "./App.css";
 import Header from "./Components/Header";
 import Home from "./Components/Home";
 import Articles from "./Components/Articles";
+import ArticleIdContext from "./Contexts/ArticleId";
+import SingleArticle from "./Components/SingleArticle";
 
 function App() {
+  const [articleId, setArticleId] = useState(null);
```

Comment:
Article Id doesn't need to be passed down via context - it can be accessed from the parameters of the endpoint and passed to comments via props (it's a simple enough relationship that context is not necessary and props would be better)